src/main/scala/com/fulcrumgenomics/umi/ExtractUmisFromBam.scala

```diff
@@ -126,10 +126,19 @@ class ExtractUmisFromBam
     if (molecularIndexTags.contains(tag)) invalid(s"$tag specified as single-tag and also per-index tag")
   }
 
-  override def execute(): Unit = {
-    assert(molecularIndexTags1.length == rs1.molecularBarcodeSegments.length)
-    rs2.foreach(rs => assert(molecularIndexTags2.length == rs.molecularBarcodeSegments.length))
+  // Validate that the # of molecular barcodes in the read structure matches the tags
+  {
+    def validateMolecularBarcodesInReadStructureAndTag(rs: ReadStructure, tags: Seq[String]): Unit = {
+      validate(molecularIndexTags1.length == rs.molecularBarcodeSegments.length,
```

Comment:
`molecularIndexTags1.length` should be `tags` here

Reply:
Also, I think `validate` only works how you want it to if it's in the constructor code.  If we want it to work the same in the execute() method then I think we need to add another case to the catch in `FgBioMain` around ~104.